### ISSUE TITLE
PSyclone : undo/redo

### ISSUE THREAD

**Issue by rupertford** [stfc/PSyclone#262]
Copied from mosrs ticket 242.

undo/redo would be useful if optimising interactively in a gui.

undo/redo works in theory, but in practice it fails with recursion errors as we use a deepcopy of the objects and these copy more than we need (particularly the code itself).
undo/redo has been switched off in #197 (the deepcopy's have been commented out) and this will go on the trunk in the near future.
A solution needs to be found. A potential solution is a custom deep copy which skips the code and leaves that as a reference.

**Comment by LonelyCat124** [comment]
@sergisiso @arporter I think we've moved further away from this being possible? Is this a wontfix and close issue at this point?

**Comment by sergisiso** [comment]
Yeah, in a sense we already have deepcopies and a gui could use that to manage its interactive undo operations. But it is not a goal to have psyir/transformation undo themself.
